calyx-py/test/correctness/matrix.py

```diff
@@ -1,107 +1,128 @@
 import calyx.builder as cb
 
+
 def insert_matmul_component(prog, n):
     """Inserts the component `matmul` into the program.
 
-    It has: 
+    It has:
     - one 2d combinational ref memory, A
     - two 2d sequential ref memories, B and C
 
-    Interpreting A and B as n x n matrices, matmul computes the matrix product 
+    The memories are of a variety of flavors just to demonstrate what is similar
+    and different between them; there is not a deep reason we have chosen these.
+
+    By and large we recommend the use of sequential memories, which are more realistic.
+
+    Interpreting A and B as n x n matrices, matmul computes the matrix product
     A*B and writes this into C.
     """
 
-    logn = n.bit_length() 
-    
+    logn = n.bit_length()
+
     matmul = prog.component("matmul")
-    
-    # matrices
+
+    # The matrices are declared using a similar syntax.
     A = matmul.comb_mem_d2("A", 32, n, n, logn, logn, is_ref=True)
-    B = matmul.seq_mem_d2( "B", 32, n, n, logn, logn, is_ref=True)
-    C = matmul.seq_mem_d2( "C", 32, n, n, logn, logn, is_ref=True)
+    B = matmul.seq_mem_d2("B", 32, n, n, logn, logn, is_ref=True)
+    C = matmul.seq_mem_d2("C", 32, n, n, logn, logn, is_ref=True)
 
     mult = matmul.mult_pipe(32)
     add = matmul.add(32)
-
     acc = matmul.reg(32)
 
-    # iterators: i, j, k ∈ [0, n)
+    # Iterators: i, j, k ∈ [0, n)
     i = matmul.reg(3)
     j = matmul.reg(3)
     k = matmul.reg(3)
 
-    # matrix entries
-    a = matmul.reg(32) 
-    b = matmul.reg(32) 
-
-
-    zero_acc = matmul.reg_store(acc, 0) # acc := 0
-    zero_i = matmul.reg_store(i, 0)     # i := 0
-    zero_j = matmul.reg_store(j, 0)     # j := 0
-    zero_k = matmul.reg_store(k, 0)     # k := 0
-
-    cond_i = matmul.lt_use(i.out, n)    # i < n
-    cond_j = matmul.lt_use(j.out, n)    # j < n
-    cond_k = matmul.lt_use(k.out, n)    # k < n
-
-    read_A = matmul.mem_load_d2(A, i.out, k.out, a, "read_A") # a := A[i][k]
-    read_B = matmul.mem_load_d2(B, k.out, j.out, b, "read_B") # b := B[k][j]
-    
-    # C[i][j] := c
-    write_C = matmul.mem_store_d2(C, i.out, j.out, acc.out, "write") 
-   
-    # acc := acc + (a * b)
-    with matmul.group("upd") as upd:
-        # compute a*b
+    # A register to store an entry of matrix A. We won't need one for matrix B.
+    a = matmul.reg(32)
+
+    zero_acc = matmul.reg_store(acc, 0)  # acc := 0
+    zero_j = matmul.reg_store(j, 0)  # j := 0
+    zero_k = matmul.reg_store(k, 0)  # k := 0
+
+    cond_i = matmul.lt_use(i.out, n)  # i < n
+    cond_j = matmul.lt_use(j.out, n)  # j < n
+    cond_k = matmul.lt_use(k.out, n)  # k < n
+
+    # a := A[i][k]
+    load_A = matmul.mem_load_d2(A, i.out, k.out, a, "read_A")
+    # Latch B[k][j], so that we can later read `B.read_data` and get B[k][j].
+    # While `mem_load` works on combinational and sequential memories,
+    # `mem_latch` only works on combinational memories.
```

Comment:
```suggestion
    # `mem_latch` only works on sequential memories.
```
Based on [this](https://github.com/calyxir/calyx/pull/2172/files#diff-1b6ee22a9e8b557927fd938de3186a9db39b2efd86dd63605f498edeb3d75160R871)?

Reply:
You were right, and the code was right. The comment was lying! Thanks for the catch, now fixed.